### PR TITLE
Add push-based widget updates

### PR DESCRIPTION
**Pull request by hariharanjagan**
## Summary

Adds iOS 26 WidgetKit push support so widgets refresh via push instead of only
on the system's timeline schedule.

Using the new iOS 26 `WidgetPushHandler` API, a `HAWidgetPushHandler` is
attached to the data widgets. When the system issues a widget push token, the
handler registers that token together with the widget's tracked `entity_ids`
with the connected Home Assistant server, via the new `register_push_subscription`
webhook command. When any of those entities change, the server sends an APNs
`widgets` push and WidgetKit reloads the widget's timeline — no polling.

Covered widgets (those with a fixed, configured entity list):
- **Sensors** — the entities chosen in the widget
- **To-do list** — the selected list entity
- **Custom** — the entity-type items in the widget

Deliberately excluded:
- **Gauge / Details** — template-driven (Jinja rendered server-side), so there
  is no fixed entity list to subscribe to
- **Commonly used** — entities are computed dynamically, so a static
  subscription would go stale

Also fixes a build break in `WidgetSensors` whose `body` bound the configuration
to `let config` without returning it, which prevented the Widgets extension from
compiling.

## Screenshots

No new UI — this is a background refresh mechanism. Existing widgets simply
reload their content sooner (on entity change) instead of waiting for the next
scheduled timeline reload.

After Widget Push Notification
IMG_3132

Before Widget Push Notification
IMG_3131

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#

## Any other notes

This is the iOS portion of a three-part feature:
1. **Home Assistant Core** — the platform-agnostic push-subscription mechanism
   (`register_push_subscription` webhook + entity-change tracking):
   home-assistant/core#174943
2. **This PR** — the iOS `WidgetPushHandler` that registers the widget push
   token and tracked entities
3. **Push relay** — formats the outgoing APNs `widgets` push from the
   subscription payload (separate change)

 **The Widgets extension needs the Push Notifications capability
(`aps-environment`) added** for the system to issue a widget push token. I left
that out of this PR because it's signing/capability configuration rather than
code — flagging it for the mobile team to add.

Validated end-to-end on a real iOS 26.5 device for the Sensors widget: the
handler registered the token + entities, an entity change triggered an APNs
`widgets` push, and the widget reloaded with the new value. To-do and Custom
compile and follow the identical registration path.

### Related Pull Requests
Core - https://github.com/home-assistant/core/pull/174943
FCM - https://github.com/home-assistant/mobile-apps-fcm-push/pull/337

Note: the fork's distribute/CI job fails at code signing (it needs the repo's
signing secrets, which forks don't have) — not a code issue.

### CODE REVIEW

File: Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        let subscriptionID: String
+        let target: String
+        let entityIDs: Set<String>
+    }
+
+    func pushTokenDidChange(_ pushInfo: WidgetPushInfo, widgets: [WidgetInfo]) {
+        let tokenHex = pushInfo.token.map { String(format: "%02x", $0) }.joined()
+        let subscriptions = Self.subscriptions(from: widgets).filter { !$0.entityIDs.isEmpty }
+
+        Current.Log.info("HAWidgetPushHandler: token \(tokenHex.prefix(8))… -> \(subscriptions.count) subscription(s)")
```
[the remark this answers is not in the document]
Comment (hariharanjagan) [reply Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift]: Fixed in 58fd51a2e — the log no longer includes the token; it now logs only the subscription count/context.

File: Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    "HAWidgetPushHandler: registering \(subscription.subscriptionID) " +
+                        "(\(subscription.entityIDs.count) entities) with \(server.identifier.rawValue)"
+                )
+                Task {
+                    try? await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+                        Current.webhooks.sendEphemeral(server: server, request: request)
+                            .done { continuation.resume() }
+                            .catch { continuation.resume(throwing: $0) }
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (hariharanjagan) [reply Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift:61]: Fixed in 58fd51a2e — the webhook send is now awaited in a do/catch and failures are logged via Current.Log.error instead of try?.

File: Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+//  HomeAssistant
+//
+//  Created by Hariharan on 29/06/26.
+//  Copyright © 2026 Home Assistant. All rights reserved.
+//
+
+
+import Shared
+import WidgetKit
+
```
[the remark this answers is not in the document]
Comment (hariharanjagan) [reply Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift]: Addressed in 58fd51a2e.

File: Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                    ]
+                )
+                Current.Log.info(
+                    "HAWidgetPushHandler: registering \(subscription.subscriptionID) " +
+                        "(\(subscription.entityIDs.count) entities) with \(server.identifier.rawValue)"
+                )
+                Task {
+                    do {
+                        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+                            Current.webhooks.sendEphemeral(server: server, request: request)
```
Comment (bgoncal) [inline Sources/Extensions/Widgets/Sensor/HAWidgetPushHandler.swift:51]: If this fails, when will it be retried?

**Comment by bgoncal** [comment]
Keeping it draft while we dont have the core PR ready, since all depends on that merged first

**Comment by bgoncal** [comment]
Please in all PR's you opened add a footer with link to all related PRs:
- Core
- iOS
- FCM

So it's easier to navigate between them

**Comment by gbougakov** [comment]
Hi! I was investigating implementing the new iOS 27 Now Playing API (specifically the [remote sessions API](https://developer.apple.com/documentation/nowplaying/publishing-remote-media-sessions)), and this PR + the corresponding PR in core would make it a lot easier to implement :) 

I was thinking of implementing it but the task of shepherding three PRs through review feels a bit daunting, however if this merges, Now Playing could be added with just a small PR to FCM and companion.

If I can assist in any way to help push these PRs through, please tell — I would be happy to help

**Comment by bgoncal** [comment]
Unfortunately the bottleneck has been my availability to dedicate focus on reviewing core PR and testing the implementation + edge cases.

If you want to help out perhaps you can test the core PR, explore possible edge cases, comment your findings there and review it of course, it's on my list to pick this up but unfortunately no ETA.

**Comment by bgoncal** [comment]
@gbougakov I spent some tokens on initial testing here https://github.com/home-assistant/core/pull/174943#issuecomment-5410471128
